src/core/credentials/transport/tls/tls_security_connector.cc

```diff
@@ -584,16 +608,23 @@ TlsServerSecurityConnector::TlsServerSecurityConnector(
     watched_identity_cert_name = options_->identity_cert_name();
   }
   // Server side won't use default system roots at any time.
-  distributor->WatchTlsCertificates(std::move(watcher_ptr),
-                                    watched_root_cert_name,
-                                    watched_identity_cert_name);
+  if (watched_root_cert_name.has_value()) {
```

Comment:
Root certs will always be watched on the client side, so there's no need for this `if` -- we should unconditionally start the watch for root certs.